### PR TITLE
Navigation and Root UI tests

### PR DESCRIPTION
**Pull request by kacperoak**
- Clean up navigation
- Add UI test for Web3ModalRootState, Web3ModalRoot and Web3ModalTopBar

### CODE REVIEW

File: product/web3modal/src/androidTest/kotlin/com/walletconnect/web3/modal/ui/components/internal/root/Web3ModalRootTest.kt [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                controller.navigate(Route.REDIRECT.path + "&" + "title")
+            }
+        }
+
+        composeTestRule.onNodeWithTag("Title").assertExists()
+        composeTestRule.onNodeWithText("title").assertExists()
+    }
+
+    @Test
+    fun web3ModalRoot_shouldWithoutTitle() {
```
Comment (Elyniss) [inline product/web3modal/src/androidTest/kotlin/com/walletconnect/web3/modal/ui/components/internal/root/Web3ModalRootTest.kt:57]:
```suggestion
    fun web3ModalRoot_shouldShowWithoutTitle() {
```

**Review by Elyniss — APPROVED** [review]

File: product/web3modal/src/androidTest/kotlin/com/walletconnect/web3/modal/ui/TestUtils.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import androidx.navigation.createGraph
+import androidx.navigation.testing.TestNavHostController
+import com.walletconnect.web3.modal.ui.navigation.Route
+import com.walletconnect.web3.modal.ui.navigation.addTitleArg
+
+@Composable
+fun rememberTestNavController(): TestNavHostController {
+    val context = LocalContext.current
+    return remember {
+        TestNavHostController(context).apply {
```
Comment (TalhaAli00) [inline product/web3modal/src/androidTest/kotlin/com/walletconnect/web3/modal/ui/TestUtils.kt:18]:
```suggestion
        TestNavHostController(LocalContext.current).apply {
```

Comment (TalhaAli00) [inline product/web3modal/src/androidTest/kotlin/com/walletconnect/web3/modal/ui/TestUtils.kt:18]:
```suggestion
    return remember {
        TestNavHostController(LocalContext.current).apply {
```

File: product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/components/internal/root/Web3ModalRootState.kt [hunk 3]
```diff
@@ -28,16 +25,15 @@ internal class Web3ModalRootState(
...
 ) {
-    private val currentDestinationFlow: Flow<NavBackStackEntry>
+    val currentDestinationFlow: Flow<NavBackStackEntry>
         get() = navController.currentBackStackEntryFlow
 
     val canPopUp: Boolean
         get() = !topLevelDestinations.any { it.path == navController.currentDestination?.route }
 
-    val title: StateFlow<String?>
+    val title: Flow<String?>
```
Comment (TalhaAli00) [inline product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/components/internal/root/Web3ModalRootState.kt:34]: why switch from StateFlow to Flow?

File: product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/routes/connect/all_wallets/AllWalletsRoute.kt [hunk 4]
```diff
@@ -56,7 +54,7 @@ internal fun AllWalletsRoute(
 ) {
     AllWalletsContent(
         wallets = wallets,
-        onWalletItemClick = { navController.navigate(Route.REDIRECT.path + "/${it.id}&${it.name}") },
+        onWalletItemClick = { navController.navigateToRedirect(it) },
```
Comment (TalhaAli00) [inline product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/routes/connect/all_wallets/AllWalletsRoute.kt:57]: it's add an explicit name for it

**Review by TalhaAli00 — APPROVED** [review]

**Pull request merged**